wellcomeml/ml/clustering.py

```diff
@@ -209,17 +233,22 @@ def optimise(self, X, param_grid, n_cluster_range=None, max_noise=0.2,
                 grid.cv_results_['mean_test_noise'],
                 grid.cv_results_['mean_test_n_clusters']
         ):
+
             if min_n_clusters <= n_clusters <= max_n_clusters\
                     and noise <= max_noise\
-                    and silhouette > best_silhouette:
+                    and silhouette >= best_silhouette:
                 best_silhouette = silhouette
                 best_params = params
```

Comment:
perhaps it would be useful to have a log comment to explain if there are no parameters that satisfy the conditions? I know this occurred when I was using it where I had to check that best_params={} to explain why my model hadn't changed after optimising

Reply:
Good idea. Added in https://github.com/wellcometrust/WellcomeML/pull/148/commits/0395a0b1ad70ca84cfe0aa992c621f06551c5cd7